Documentation/botr/ryujit-overview.md

```diff
@@ -387,6 +382,37 @@ The RyuJIT register allocator uses a Linear Scan algorithm, with an approach sim
 * `Intervals` (representing live ranges of variables or tree expressions) and `RegRecords` (representing physical registers), both of which derive from `Referenceable`.
 * `RefPositions`, which represent uses or defs (or variants thereof, such as ExposedUses) of either `Intervals` or physical registers.
 
+`LinearScan::buildIntervals()` traverses the entire method building RefPositions and Intervals as required. For example, for the `STORE_BLK` node in this snippet:
+```
+t67 =    CNS_INT(h) long   0x2b5acef2c50 static Fseq[s1]
+     ┌──▌  t67    long   
+ t0 = ▌  IND       ref
+ t1 =    CNS_INT   long   8 Fseq[#FirstElem]
+     ┌──▌  t0     ref    
+     ├──▌  t1     long   
+ t2 = ▌  ADD       byref
+     ┌──▌  t2     byref  
+ t3 = ▌  IND       struct
+t31 =    LCL_VAR_ADDR byref  V08 tmp1
+     ┌──▌  t31    byref  
+     ├──▌  t3     struct 
+      ▌  STORE_BLK(40) struct (copy) (Unroll)
+```
+the following RefPositions are generated:
+```
+N027 (???,???) [000085] -A-XG-------              ▌  STORE_BLK(40) struct (copy) (Unroll) REG NA
+Interval 16: int RefPositions {} physReg:NA Preferences=[allInt]
+<RefPosition #40  @27  RefTypeDef <Ivl:16 internal> STORE_BLK BB01 regmask=[allInt] minReg=1>
+Interval 17: float RefPositions {} physReg:NA Preferences=[allFloat]
+<RefPosition #41  @27  RefTypeDef <Ivl:17 internal> STORE_BLK BB01 regmask=[allFloat] minReg=1>
+<RefPosition #42  @27  RefTypeUse <Ivl:15> BB01 regmask=[allInt] minReg=1 last>
+<RefPosition #43  @27  RefTypeUse <Ivl:16 internal> STORE_BLK BB01 regmask=[allInt] minReg=1 last>
+<RefPosition #44  @27  RefTypeUse <Ivl:17 internal> STORE_BLK BB01 regmask=[allFloat] minReg=1 last>
+    +<TreeNodeInfo 0=1 1i 1f src=[allInt] int=[rax rcx rdx rbx rbp rsi rdi r8-r15 mm0-mm15] dst=[allInt] I>[--]
+  consume= 1 produce=0
+```
+The “@ 27” is the location number of the node. “internal” indicates a register that is internal to the node (in this case 2 internal registers are needed, one float (XMM on XARCH) and one int, as temporaries for copying). “regmask” indicates the register constraints for the `RefPosition`.
```

Comment:
@CarolEidt  I removed any description related to `TreeNodeInfo` even if it's still in the dump. It looks rather useless and redundant to me, perhaps we should just remove it from the dump? Everything is about refpositions and intervals now.

Reply:
I'd left the `TreeNodeInfo` dumps in originally in order to make the diffs easier, but that was long ago, so definitely time to clean it up in teh actual dumps (though that can obviously be deferred to a future PR).

Reply:
I'll take a look, probably it's easy to cleanup and then it doesn't worth a separate PR just for removing some dump code.

Reply:
Hmm, I was checking to see if there's anything we would lose by simply deleting the `TreeNodeInfo` message and noticed something odd about the `RefPosition` dump. The `<RefPosition...` message is generated at the end of `newRefPosition` but in many cases the caller will modify the ref position after creating it:
```C++
RefPosition* defRefPosition =
    newRefPosition(interval, currentLoc + 1, RefTypeDef, tree, dstCandidates, multiRegIdx);
if (tree->IsUnusedValue())
{
    defRefPosition->isLocalDefUse = true;
    defRefPosition->lastUse       = true;
}
```
It took me a while to understand why "local" wasn't appearing in the dump. Well, "local" appears but only in the subsequent ref position dumps. Not sure what would be the best way to avoid this confusing behavior. The easiest way would be for the caller to do the dump when it has finished setting up the ref position but there's a good chance that we'll forget to dump when we add new ref position creating code in the future.

Reply:
I removed the entire `dumpNodeInfo` function. Might be debatable as some of the information it shows isn't readily available somewhere else, `srcCount` and `dstCount` in particular. But those are IMO weird, some double bookkeeping that just complicated things.

Reply:
Yes, there's probably a better approach to sanity checking.